src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -146,17 +153,89 @@ def get_nodepool_name(self) -> str:
 
         # this parameter does not need dynamic completion
         # validation
-        instances = cf_agent_pools.list(self.get_resource_group_name, self.get_cluster_name)
-        for agentpool_profile in instances:
-            if agentpool_profile.name == nodepool_name:
-                raise InvalidArgumentValueError(
-                    "Node pool {} already exists, please try a different name, "
-                    "use 'aks nodepool list' to get current list of node pool".format(
-                        nodepool_name
+        if enable_validation:
+            instances = cf_agent_pools.list(self.get_resource_group_name, self.get_cluster_name)
```

Comment:
just curious, will "list" raise any Exception?

Reply:
Yes, it might raise an exception and I am not sure if there is any other choice than raise the exception directly

Reply:
I mean do we handle that exceptions in our code? or just let the cli crash?

Reply:
We did not handle potential exceptions before, and yes, cli would exit with the error message.

If this is an error returned by rp, the error type is probably `HttpResponseError`, and it contains relatively complete error information. I suppose additional processing is not necessary?